etc/schema.json

```diff
@@ -172,19 +172,50 @@
             }
         },
         "app_proto": {
-            "type": "string"
+            "type": "string",
+            "description": "Application layer protocol of the flow",
+            "suricata": {
+                "keywords": [
+                    "app-layer-protocol"
+                ]
+            }
         },
         "app_proto_expected": {
-            "type": "string"
+            "type": "string",
+            "description": "In case of a protocol change to a specific protocol, and this specific protocol was not recognised, this field will have the value of the expected protocol",
+            "suricata": {
+                "keywords": []
+            }
         },
         "app_proto_orig": {
-            "type": "string"
+            "type": "string",
+            "description": "Original application layer protocol of the flow after a protocol change",
+            "suricata": {
+                "usage": "the keyword should be used with its `original` option like `app-layer-protocol:xyz,original`",
+                "keywords": [
+                    "app-layer-protocol"
+                ]
+            }
         },
         "app_proto_tc": {
-            "type": "string"
+            "type": "string",
+            "description": "Application layer protocol detected to client in case of mismatch",
+            "suricata": {
+                "usage": "Usage: the keyword should be used with its `to_client` option like `app-layer-protocol:xyz,to_client`",
+                "keywords": [
+                    "app-layer-protocol"
+                ]
+            }
         },
         "app_proto_ts": {
-            "type": "string"
+            "type": "string",
+            "description": "Application layer protocol detected to server in case of mismatch",
+            "suricata": {
+                "usage": "Usage: the keyword should be used with its `to_server` option like `app-layer-protocol:xyz,to_server`",
```

Comment:
Can we leave this for another task, where we actually implement usage somehow? Right now we have this here, as well as the documentation, it would be nice to get to one source of truth.  But if you must, merge into description.  Maybe start a ticket, documented how you think this new field should be used?

Reply:
> where we actually implement usage somehow?

What is there to implement ?

I just want some free-form text to add details about detection on the field getting logged...

> Right now we have this here, as well as the documentation

I do not think we have this in the documentation : by this, I mean the link from a field getting logged to how it can be detected (when it is more complex than just using the right sticky buffer keyword)

> But if you must, merge into description

I do not like that because I understand `description` being about the eve.json and I want a separate field to tell about rules...

Maybe I will do that anyways...